git-keeper-server/gkeepserver/create_user.py

```diff
@@ -231,3 +232,29 @@ def create_student_user(student: Student):
                 student.first_name, student.last_name,
                 email_address=student.email_address,
                 additional_groups=groups, shell='git-shell')
+
+
+def add_faculty(last_name, first_name, email_address, admin=False):
+    """
+    Add a faculty user.
+
+    :param last_name: last name of the faculty member
+    :param first_name: first name of the faculty member
+    :param email_address: email address of the faculty member
+    :param admin: True if the faculty member should be an admin
+    """
+
+    username = db.insert_user(email_address, first_name, last_name,
+                              'faculty')
+
+    if admin:
+        db.set_admin(username)
+
+    gkeepd_logger.log_info('Adding faculty user {}'.format(username))
+
+    groups = [config.keeper_group, config.faculty_group]
+
+    create_user(username, UserType.faculty, first_name, last_name,
```

Comment:
The `email_address` and `additional_groups` parameters are optional. If the email address is provided it will send an email to the user, and if additional groups are provided then the user will be added to those groups.